test/nodes/test_doc_language_classifier.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+import logging
+
+from haystack.schema import Document
+from haystack.nodes.doc_language_classifier import (
+    LangdetectDocumentLanguageClassifier,
+    TransformersDocumentLanguageClassifier,
+)
+
+DOCUMENTS = [
+    Document(content="My name is Matteo and I live in Rome"),
+    Document(content="Mi chiamo Matteo e vivo a Roma"),
+    Document(content="Mi nombre es Matteo y vivo en Roma"),
+]
+
+EXPECTED_LANGUAGES = ["en", "it", "es"]
+
+
+@pytest.fixture(params=["langdetect", "transformers"])
+def doclangclassifier(request):
+    if request.param == "langdetect":
+        return LangdetectDocumentLanguageClassifier(route_by_language=True, languages_to_route=["en", "es", "it"])
+    elif request.param == "transformers":
+        return TransformersDocumentLanguageClassifier(route_by_language=True, languages_to_route=["en", "es", "it"])
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("doclangclassifier", ["langdetect", "transformers"], indirect=True)
+def test_doclangclassifier_predict(doclangclassifier):
+    results = doclangclassifier.predict(documents=DOCUMENTS)
+    for doc, expected_language in zip(results, EXPECTED_LANGUAGES):
+        assert doc.to_dict()["meta"]["language"] == expected_language
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("doclangclassifier", ["langdetect", "transformers"], indirect=True)
+def test_doclangclassifier_predict_batch(doclangclassifier):
+    results = doclangclassifier.predict_batch(documents=[DOCUMENTS, DOCUMENTS[:2]])
+    expected_languages = [EXPECTED_LANGUAGES, EXPECTED_LANGUAGES[:2]]
+    for lst_docs, lst_expected_languages in zip(results, expected_languages):
+        for doc, expected_language in zip(lst_docs, lst_expected_languages):
+            assert doc.to_dict()["meta"]["language"] == expected_language
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("doclangclassifier", ["langdetect", "transformers"], indirect=True)
+def test_doclangclassifier_run_not_route(doclangclassifier):
+    doclangclassifier.route_by_language = False
+    results, edge = doclangclassifier.run(documents=DOCUMENTS)
+    assert edge == "output_1"
+    for doc, expected_language in zip(results["documents"], EXPECTED_LANGUAGES):
+        assert doc.to_dict()["meta"]["language"] == expected_language
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("doclangclassifier", ["langdetect", "transformers"], indirect=True)
+def test_doclangclassifier_run_route(doclangclassifier):
+    for doc, expected_language in zip(DOCUMENTS, EXPECTED_LANGUAGES):
+        result, edge = doclangclassifier.run(documents=[doc])
+        document = result["documents"][0]
+
+        assert edge == doclangclassifier._get_edge_from_language(expected_language)
```

Comment:
Smart one, but maybe too smart :smile: 

I have mixed feelings about this one because if `_get_edge_from_language` would have any bugs, we wouldn't spot it in this test. Let's use a hardcoded list.